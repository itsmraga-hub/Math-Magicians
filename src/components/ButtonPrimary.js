import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';

class ButtonPrimary extends React.PureComponent {
  // getButtonValue = (e) => {
  //   console.log(e.target.innerText);
  //   return e.target.innerText;
  // }

  render() {
    const { name, getButtonValue } = this.props;
    return (
      <button type="button" onClick={getButtonValue} className={`btn btn-primary grid-item ${+name === 0 ? 'span-2' : null}`} data-id="" data-attribute="">{ name }</button>
    );
  }
}

ButtonPrimary.propTypes = {
  name: PropTypes.string.isRequired,
  getButtonValue: PropTypes.func.isRequired,
};

export default ButtonPrimary;
