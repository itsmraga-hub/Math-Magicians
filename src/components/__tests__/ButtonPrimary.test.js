import React from 'react';
import renderer from 'react-test-renderer';

import ButtonPrimary from '../ButtonPrimary';

const obj = {
  name: '+',
  getButtonValue: () => true,
};

it('<ButtonPrimary /> Renders properly', () => {
  const { name, getButtonValue } = obj;
  const tree = renderer.create(<ButtonPrimary
    name={name}
    getButtonValue={getButtonValue}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
