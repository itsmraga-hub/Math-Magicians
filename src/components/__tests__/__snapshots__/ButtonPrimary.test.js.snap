// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPrimary /> Renders properly 1`] = `
<button
  className="btn btn-primary grid-item null"
  data-attribute=""
  data-id=""
  onClick={[Function]}
  type="button"
>
  +
</button>
`;
